### PR TITLE
[fix](show) show create table show index comment error

### PR DESCRIPTION
**Pull request by zgxme**
## Proposed changes
Fix two bugs that cause show create table index comment error

1. comment has extra single quotes.
2. quota does not escape in show create table result.
```
COMMENT ''test index" comment'' to COMMENT 'test index\" comment'
```

Refs PR:
https://github.com/apache/doris/pull/6560
https://github.com/apache/doris/pull/28346

### CODE REVIEW

**Comment by zgxme** [comment]
run buildall

**Comment by zgxme** [comment]
run buildall

**Comment by zgxme** [comment]
run buildall

**Comment by zgxme** [comment]
run performance

**Comment by zgxme** [comment]
run p0

**Comment by zgxme** [comment]
run p0

**Comment by zgxme** [comment]
run buildall

**Comment by morrySnow** [comment]
replace screenshot with text for git msg and search engine firendly

**Comment by zgxme** [comment]
@morrySnow ok, done

**Comment by morrySnow** [comment]
> @morrySnow ok, done

still have screenshot in description

**Comment by zgxme** [comment]
@morrySnow  removed.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java [hunk 1]
```diff
@@ -2753,7 +2753,9 @@ public IndexDefinition visitIndexDef(IndexDefContext ctx) {
         List<String> indexCols = visitIdentifierList(ctx.cols);
         Map<String, String> properties = visitPropertyItemList(ctx.properties);
         String indexType = ctx.indexType != null ? ctx.indexType.getText().toUpperCase() : null;
-        String comment = ctx.comment != null ? ctx.comment.getText() : "";
+        //comment should remove '\' and '(") at the beginning and end
+        String comment = ctx.comment != null ? ctx.comment.getText().substring(1, ctx.comment.getText().length() - 1)
+                .replace("\\", "") : "";
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java:2758]: String comment = ctx. comment == null ? "" : LogicalPlanBuilderAssistant.escapeBackSlash(
                ctx. comment.getText().substring(1, ctx.STRING_LITERAL().getText().length() - 1));

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Index.java [hunk 2]
```diff
@@ -17,18 +17,21 @@
 
 package org.apache.doris.catalog;
 
+
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Index.java:20]: remove this line

**Comment by zgxme** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Index.java (hunk 2 above)
Comment (zgxme) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Index.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java (hunk 1 above)
Comment (zgxme) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java]: done

**Review by morrySnow — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Comment by zgxme** [comment]
@morrySnow May need to remove dev/2.0 tags, doesn't seem to affect 2.0？

**Pull request merged**